src/renderer/rendered_window.rs

```diff
@@ -453,12 +507,37 @@ impl RenderedWindow {
                 line_fragments,
             } => {
                 tracy_zone!("draw_line_cmd", 0);
-
-                self.actual_lines[row] = Some(Arc::new(Mutex::new(Line {
+                // At the moment we only consider top and bottom borders, and winbar is also one type of border
+                // So, lines with purly border highlight groups are considered borders.
+                let is_border = line_fragments
+                    .iter()
+                    .map(|fragment| {
+                        fragment.style.as_ref().map_or(false, |style| {
+                            style.infos.last().map_or(false, |info| {
+                                [
+                                    "FloatBorder",
+                                    "FloatTitle",
+                                    "FloatFooter",
+                                    "WinBar",
+                                    "WinBarNC",
+                                ]
+                                .iter()
+                                // The specification seems to indicate that kind should be UI and
+                                // then we only need to test ui_name. But at least for FloatTitle,
+                                // that is not the case, the kind is set to syntax and hi_name is
+                                // set.
+                                .map(|v| info.ui_name == *v || info.hi_name == *v)
+                                .any(|v| v)
+                            })
+                        })
+                    })
+                    .all(|v| v);
```

Comment:
```suggestion
                    .any(|v| v);
```

> lines with purly border highlight groups are considered borders.

I believe if we change this to "lines **contain** border highlight grouops are considered borders", then dropbar.nvim will work! Don't know if this has false positives. @fredizzimo what do you think?

Reply:
However with this change, it seems that we need another way to ignore left and right borders.

Reply:
That's a great idea!

We could use all to detect the borders and any to detect winbars. I will try that later today.

Reply:
I implemented this, and it does seem to work. 

But I have only done a very quick test yet and it's a bit late for me to do more testing now.